ml-agents/mlagents/trainers/settings.py

```diff
@@ -215,6 +215,12 @@ def structure(d: Mapping, t: type) -> Any:
         return d_final
 
 
+@attr.s(auto_attribs=True)
+class ExtrinsicSettings(RewardSignalSettings):
+    # For use with COMA2. Add groupmate rewards to the final extrinsic reward.
+    add_groupmate_rewards = False
```

Comment:
Is this a new hyperparameter for the user ?
Looks like the value is hard coded anyway in the coma trainer, is it really necessary ?

Reply:
It is not a new hyperparameter and it is actually to avoid requiring users to learn about a new group reward signal. POCA will set this to True on initialization so that the ExtrinsicReward provider knows to collect teammate rewards. Otherwise it will be False. The only way for users to change it is to modify python.

Reply:
Then why is it a setting and not a constant in the code?

Reply:
The alternative would be to have a public variable that is set later (ew) or additional constructor parameter to Extrinsic Reward Provider (and only Extrinsic Reward Provider). This would mess with the BaseRewardProvider abstraction. I don't really have a preference, but it is choosing between the lesser of several evils here.

Reply:
Potential alternative: #5082